domain/relation/service/relation.go

```diff
@@ -314,13 +336,38 @@ func (s *Service) GetRelationUUIDByID(ctx context.Context, relationID int) (core
 	return s.st.GetRelationUUIDByID(ctx, relationID)
 }
 
-// GetRelationUUIDFromKey returns a relation UUID for the given Key.
+// GetRelationUUIDByKey returns a relation UUID for the given Key.
 //
 // The following error types can be expected:
 //   - [relationerrors.RelationNotFound]: when no relation exists for the given
 //     key.
-func (s *Service) GetRelationUUIDFromKey(ctx context.Context, relationKey corerelation.Key) (corerelation.UUID, error) {
-	return "", coreerrors.NotImplemented
+func (s *Service) GetRelationUUIDByKey(ctx context.Context, relationKey corerelation.Key) (corerelation.UUID, error) {
+	endpointIdentifiers, err := internalrelation.ParseRelationKey(relationKey)
```

Comment:
ParseRelationKey should be local to the domain. It has no use outside of it.

Rename to `ParseRelationKeyEndpoints`, to disambiguate from ParseKeyFromTagString which has a very different result.